File: q0242/q0242.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length())
            return false;

        int[] characterMapS = new int[26];
        int[] characterMapT = new int[26];
        for (int i = 0; i < s.length(); i++) {
            char charS = s.charAt(i);
            char charT = t.charAt(i);
            characterMapS[charS - 'a']++;
            characterMapT[charT - 'a']++;
        }

        return Arrays.equals(characterMapS, characterMapT);
    }
}

File: q0242/q0242.cpp
class Solution
{
public:
    bool isAnagram(string s, string t)
    {
        if (s.size() != t.size())
            return false;

        vector<int> characterMapS(26);
        vector<int> characterMapT(26);
        for (int i = 0; i < s.size(); i++)
        {
            characterMapS[s[i] - 'a']++;
            characterMapT[t[i] - 'a']++;
        }

        return characterMapS == characterMapT;
    }
};